Automated backport of #967: Fix flaky e2e failures with kindnet cni

tpantelis opened the pull request:
Backport of #967 on release-0.13.

#967: Fix flaky e2e failures with kindnet cni

For details on the backport process, see the [backport requests](https://submariner.io/development/backports/) page.

dfarrell07 commented:
[Shell](https://github.com/submariner-io/shipyard/actions/runs/3235315942/jobs/5299574639#logs)

```
In scripts/shared/clusters.sh line 17:
source "${SCRIPTS_DIR}/lib/shflags"
       ^--------------------------^ SC1091 (info): Not following: ./lib/shflags: openBinaryFile: does not exist (No such file or directory)
```

Not sure what's going on with these failures.

tpantelis committed:
Fix flaky e2e failures with kindnet cni

While transitioning to kindnet CNI in our KIND deployments,
it was seen that HostNetworking e2e tests were occasionally
failing. On debugging this issue, it was found that kindnet
CNI does not create a dedicated CNI Interface on the nodes.
Submariner relies on the CNI Interface to support some of
the use-cases like Health-check between the GW nodes and
HostNetwork to remote cluster support. However, it was seen
that as soon as we schedule a pod on the node, kindnet CNI
creates a veth-xxx interface on the Host which can be used
as a CNI Interface. So, in order to continue validating the
above Submariner use-cases, this PR schedules a dummypod as
a DaemonSet in the clusters as a workaround (and this can be
removed when we switch to a different CNI).

Fixes: #966
Related to: #858

Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>
Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

tpantelis commented:
> [Shell](https://github.com/submariner-io/shipyard/actions/runs/3235315942/jobs/5299574639#logs)
> 
> ```
> In scripts/shared/clusters.sh line 17:
> source "${SCRIPTS_DIR}/lib/shflags"
>        ^--------------------------^ SC1091 (info): Not following: ./lib/shflags: openBinaryFile: does not exist (No such file or directory)
> ```
> 
> Not sure what's going on with these failures.

Me neither. It's happening with other PRs on the release-13 branch. I disabled the job as required.
